go_modules/lib/dependabot/go_modules/file_updater/go_mod_updater.rb

```diff
@@ -179,7 +158,16 @@ def run_go_get
 
           File.write(tmp_go_file, "package dummypkg\n") unless package
 
-          _, stderr, status = Open3.capture3(ENVIRONMENT, "go get -d")
+          # TODO: go 1.18 will make `-d` the default behavior, so remove the flag then
+          command = "go get -d"
+          # `go get` accepts multiple packages, each separated by a space
+          dependencies.each do |dep|
+            # Use version pinning rather than `latest` just in case
+            # a new version gets released in the middle of our run.
+            version = "v" + dep.version.sub(/^v/i, "")
```

Comment:
Any idea what the `version.sub(/^v/i, "")` is protecting against? Never having written Ruby before, I'm not even quite sure what this is doing... 

I copied it from https://github.com/dependabot/dependabot-core/blob/7a2085bbee797ba7ffd0c09c83fc4d5481c859c2/go_modules/lib/dependabot/go_modules/file_updater/go_mod_updater.rb#L158 but it's not used over in https://github.com/dependabot/dependabot-core/blob/7a2085bbee797ba7ffd0c09c83fc4d5481c859c2/go_modules/lib/dependabot/go_modules/update_checker.rb#L74

I'm unclear why the inconsistency between these two. It it a bug or intentional?

Reply:
It's a bit of defensive coding to make sure we never accidentally end up with a version like `vv1.1`.

Internally we store versions without the `v` prefix, so it's prepended to the version here, and in theory there might be cases where the version was parsed from a dependency file or the registry and the `v` hasn't been stripped out. I couldn't find any immediate cases where that would happen by following the code that is fed into this, but all our test-data does have the `v` prefix. 

When we run ` UpdateChecker#find_latest_resolvable_version`, that data comes from the `FileParser`, which already [strips out the `v` prefix](https://github.com/dependabot/dependabot-core/blob/7a2085bbee797ba7ffd0c09c83fc4d5481c859c2/go_modules/lib/dependabot/go_modules/file_parser.rb#L42) so it seems OK to omit it there.

It seems like the precaution it's not actually needed (anymore?), especially since `Dependabot::GoModules::Version` [strips out the prefix](https://github.com/dependabot/dependabot-core/blob/c3f01dff836d2031203c3870f772e42a42668b25/go_modules/lib/dependabot/go_modules/version.rb#L26). But, I'm always slightly nervous making these changes, as we don't have a thorough integration test-suite, and `dependency.version` can also be a straight-up string (yay, duck typing!)

I would opt to keep it in for now, the code is fairly simple and should have no meaningful performance implications. But we can investigate if we can get rid of it separately IMO.

Reply:
I'm fine either way, but the inconsistency between the two usages bothers me, because it's defensive but not... or at least appears that way. 

If I had to choose, I'd probably pick dropping it, because if Dependabot is broken for 24 hours it's not _that_ big of a deal... this isn't nuclear or hospital or banking code... and then if something does break you can track down the root cause rather than band-aiding it here.

Alternatively you could add an error log if the condition is ever true so that you defensively recover from the error, but also get notified that there's an error likely indicating a bug elsewhere in the code. But that gets a little complex, and given the impacts aren't that big of deal (as long as you've got observability metrics that alert when there's a spike in the percent of your `go_modules` jobs that are failing).

Regardless, agreed that it's out of scope for this PR, so I filed https://github.com/dependabot/dependabot-core/issues/3612 to track it.